### PR TITLE
Refresh toke, log cleanup and sign in exception handling

### PR DESCRIPTION
**Pull request by abe-mused**
Screen Shot 2022-11-08 at 01 36 56 AM
Screen Shot 2022-11-08 at 01 37 13 AM

### CODE REVIEW

**Comment by abe-mused** [comment]
PR seems pretty big, but it's because I took out all of the unnecessary log statements and added the refresh token functionality

File: lib/auth_pages/sign_in.dart [hunk 1]
```diff
@@ -22,29 +22,39 @@ class _LoginPageState extends State<LoginPage> {
...
+        if (response['status'] == true) {
           User user = response['user'];
           Provider.of<UserProvider>(context, listen: false).setUser(user);
           Navigator.pushReplacementNamed(context, '/home');
+        } else {
+          showDialog(
+              context: context,
+              builder: (context) {
+                return AlertDialog(
+                  title: const Text("Failed Login"),
```
[the remark this answers is not in the document]
Comment (abe-mused) [reply lib/auth_pages/sign_in.dart:38]: This is the title, the more meaningful message is in the next line (line 39)

File: lib/pages/community_page/community_page.dart [hunk 2]
```diff
@@ -89,127 +102,119 @@ class CommunityPageState extends State<CommunityPage> {
...
-                        style: const TextStyle(fontSize: 16),
-                        textAlign: TextAlign.center,
-                      ),
-                      const SizedBox(height: 10),
-                      Text(
-                        // ignore: unrelated_type_equality_checks
-                        "${_community.members.length} members",
-                        style: const TextStyle(fontSize: 16),
-                        textAlign: TextAlign.center,
+      body: SingleChildScrollView(
```
[the remark this answers is not in the document]
Comment (abe-mused) [reply lib/pages/community_page/community_page.dart:105]: yeah good catch, I probably missed it when I was rebasing.

File: lib/auth_pages/reset_password_code.dart [hunk 3]
```diff
@@ -73,8 +68,7 @@ class _ResetPasswordCodePageState extends State<ResetPasswordCodePage> {
             builder: (context) {
               return AlertDialog(
                 title: const Text("Success!"),
-                content: const Text(
-                    "Your password has beem reset succesfully. You can now login with your new password."),
+                content: const Text("Your password has beem reset succesfully. You can now login with your new password."),
```
Comment (mohammedali101) [inline lib/auth_pages/reset_password_code.dart:71]: been* (there's a typo here)

Comment (abe-mused) [reply lib/auth_pages/reset_password_code.dart:71]: Good catch, I'll fix it in my next PR (unless I end up having to publish another revision in this PR for another reason)

**Pull request merged**